File: minimum_number_of_vertices_to_reach_all_nodes.py
class Solution:
    def findSmallestSetOfVertices(self, n: int, edges: List[List[int]]) -> List[int]:
        indegree = [0] * n
        res = []

        for edge in edges:
            indegree[edge[1]] += 1
        
        for i in range(n):
            if indegree[i] == 0:
                res.append(i)
        
        return res

File: minimum_number_of_vertices_to_reach_all_nodes.cpp
class Solution {
public:
    vector<int> findSmallestSetOfVertices(int n, vector<vector<int>>& edges) {
        std::vector<int> indegree(n, 0);
        std::vector<int> res;

        for (const std::vector<int>& edge : edges) {
            indegree[edge[1]]++;
        }

        for (int i = 0; i < n; i++) {
            if (indegree[i] == 0) {
                res.push_back(i);
            }
        }

        return res;        
    }
};
